lib/ansible/playbook/task.py

```diff
@@ -508,6 +517,21 @@ def get_first_parent_include(self):
             return self._parent.get_first_parent_include()
         return None
 
+    def get_first_parent_block(self, static=False):
+        parent = None
+        if self._parent:
+            if isinstance(self._parent, Block):
+                if static:
+                    if getattr(self._parent, 'statically_loaded', True):
+                        parent = self._parent
+                    else:
+                        parent = self._parent.get_first_parent_block()
+                else:
+                    parent = self._parent
+            else:
+                parent = self._parent.get_first_parent_block()
+        return parent
```

Comment:
This seems suspiciously nested. How about flattening it a little?
```suggestion
        if note self._parent:
            return None

        if not isinstance(self._parent, Block):
            return self._parent.get_first_parent_block()

        if not static:
            return self._parent

        if getattr(self._parent, 'statically_loaded', True):
            return self._parent

        return self._parent.get_first_parent_block()
```

Reply:
i prefer aclear return and decision logic, having multiple returns can end up with different/incomplete returns.

Reply:
FWIW, highly nested branches are hard to follow and easier to make mistakes in.